src/main.js

```diff
@@ -17,7 +17,7 @@ const processContent = async () => {
   const applicationHealthApiURL = core.getInput('applicationHealthApiURL') || process.env.APPLICATION_HEALTH_API_URL;
   const filesPath = core.getInput('contentPath') || process.env.FILES_PATH;
   const assetsFilesPath = core.getInput('assetPath') || process.env.ASSETS_PATH;
-  const mergedAt = core.getInput('mergedAt') || Date.parse(new Date());
+  const mergedAt = Date.parse(core.getInput('mergedAt')) || Date.parse(new Date());
```

Comment:
nenhum teste?

Reply:
Então, aqui verifiquei e temos dois casos. 

o teste unitário do service.js `__tests__/service.test.js`, que tem inclusive testes que já cobriam o caso do formato errado do timestamp. E temos o teste do `main.js` ` __tests__/main.test.js`.

Aí vi que acontece o seguinte (inclusive o mesmo motivo de não ter pego um erro anteior do formato da resposta pelo service). O main está com mock em cima do `service.js`, nao da resposta do `axios`. Ou seja, a gente nao verifica em nenhum momento no teste do `main` eses caminho todo. 

Seria mais um teste `end-to-end`, onde tem que ser feito mock para tudo que é externo, e nesse caso seriam:

. mockar as respostas da api, através de um mock por cima do `axios`; 
. mockar as `actions`, (`require('@actions/core')`);

eu comecei a fazer aqui mas se mostrou bem trabalhoso, visto que tem que montar todo cenário de teste para o main rodar. 

Tive que: 

- adicionar uma lib para mockar o axios (https://github.com/ctimmerm/axios-mock-adapter);
- montar mocks para cda retorno de api (health, create_chapter e create_version); 
- vários mocks para a api das actions. 

Pra tudo funcionar, tenho que passar por todos os testes da main.js, já que algumas coisas nos testes estão quebrando e eu tenho que investigar q q está acontecendo com as promises. E promises de processos que já vimos que executa tanto em produção quanto staging. Ou seja, to dedicando tempo a entender configuração de lib pra um teste que se basea em mock. 

e confesso que nao vendo valor pra agora. To realmente preferindo isso aqui ser um débito técnico, melhoria de teste de integração/end-to-end, já que é uma action e nao um código do trybe?

Faz sentido dedicar tempo nisso agora ou adiciona uma task pra gente refatorar os testes do `main.js` ? Pergunto isso porque eu to achando muito trabalho e mais tempo dedicado pra rodar uma action em prod onde ainda estamos validando se todo o processamento está correto.

se nao fizer sentido meu questionamento, continuo aqui de boa. 

@caironoleto e @Caciquez

Reply:
Adiciona uma task então e segue bola.